src/python/pants/backend/openapi/util_rules/pom_parser.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import xml.etree.ElementTree as ET
+from dataclasses import dataclass
+
+from pants.engine.fs import Digest, DigestContents
+from pants.engine.rules import Get, collect_rules, rule
+from pants.jvm.resolve.common import Coordinate
+
+
+@dataclass(frozen=True)
+class AnalysePomRequest:
+    pom_digest: Digest
+
+
+@dataclass(frozen=True)
+class PomReport:
+    dependencies: tuple[Coordinate, ...]
+
+
+_MAVEN_NAMESPACE = "http://maven.apache.org/POM/4.0.0"
+_MAVEN_NAMESPACE_MAP = {"mvn": _MAVEN_NAMESPACE}
+
+
```

Comment:
Hah, neat. Potentially related to https://github.com/pantsbuild/pants/issues/13684